src/store/engine.ts

```diff
@@ -360,4 +364,50 @@ export const engineStore = createPartialStore<EngineStoreTypes>({
       });
     },
   },
+
+  SET_ENGINE_SUPPORTED_DEVICES: {
+    mutation(state, { engineId, supportedDevices }) {
+      state.engineSupportedDevices = {
+        ...state.engineSupportedDevices,
+        [engineId]: supportedDevices,
+      };
+    },
+  },
+
+  FETCH_AND_SET_ENGINE_SUPPORTED_DEVICES: {
+    async action({ dispatch, commit }, { engineId }) {
+      const supportedDevices = await dispatch("INSTANTIATE_ENGINE_CONNECTOR", {
+        engineId,
+      })
+        .then(
+          async (instance) =>
+            await instance.invoke("supportedDevicesSupportedDevicesGet")({})
+        )
+        .catch((e) => e);
+
+      if (!(supportedDevices instanceof Error)) {
+        commit("SET_ENGINE_SUPPORTED_DEVICES", {
+          engineId,
+          supportedDevices: supportedDevices,
+        });
+      }
+    },
+  },
```

Comment:
いろいろな考え方がるのかなと思うのですが、エラーが発生したときはエラーが発生したことがわかると良いのかなと思っています！
この流れだとfetchに失敗したときになにも起こらず、なぜか変な挙動をしてログも出ないのでわからないということになるかもです。
エラーは放っておけばロギングされる仕組みになっているので、とりあえず放置するのはどうでしょう 👀

```suggestion
  FETCH_AND_SET_ENGINE_SUPPORTED_DEVICES: {
    async action({ dispatch, commit }, { engineId }) {
      const supportedDevices = await dispatch("INSTANTIATE_ENGINE_CONNECTOR", {
        engineId,
      }).then(
        async (instance) =>
          await instance.invoke("supportedDevicesSupportedDevicesGet")({})
      );

      commit("SET_ENGINE_SUPPORTED_DEVICES", {
        engineId,
        supportedDevices: supportedDevices,
      });
    },
  },
```

Reply:
Vueってエラー投げても止まらないんですね!
と思ったら`ErrorBoundary.vue`とかいういかにもなやつに今気づきました。
修正ありがとうございます!!